cpp/src/copying/contiguous_split.cu

```diff
@@ -1158,122 +1266,742 @@ std::vector<packed_table> contiguous_split(cudf::table_view const& input,
+    std::transform(iter,
+                   iter + num_partitions,
+                   std::back_inserter(result),
+                   [&empty_inputs](int partition_index) {
+                     return packed_table{empty_inputs,
+                                         packed_columns{std::make_unique<std::vector<uint8_t>>(
+                                                          pack_metadata(empty_inputs, nullptr, 0)),
+                                                        std::make_unique<rmm::device_buffer>()}};
+                   });
+
+    return result;
+  }
+
+  std::unique_ptr<std::vector<uint8_t>> build_packed_column_metadata()
+  {
+    CUDF_EXPECTS(num_partitions == 1, "build_packed_column_metadata supported only without splits");
+
+    if (input.num_columns() == 0) { return std::unique_ptr<std::vector<uint8_t>>(); }
+
+    if (is_empty) {
+      // this is a bit ugly, but it was done to re-use make_empty_packed_table between the
+      // regular contiguous_split and chunked_pack cases.
+      auto empty_packed_tables = std::move(make_empty_packed_table().front());
+      return std::move(empty_packed_tables.data.metadata);
+    }
+
+    auto& h_dst_buf_info  = partition_buf_size_and_dst_buf_info->h_dst_buf_info;
+    auto cur_dst_buf_info = h_dst_buf_info;
+    detail::metadata_builder mb{input.num_columns()};
+
+    populate_metadata(input.begin(), input.end(), cur_dst_buf_info, mb);
+
+    return std::make_unique<std::vector<uint8_t>>(std::move(mb.build()));
   }
-  return result;
+
+  std::vector<packed_table> make_packed_tables()
+  {
+    if (input.num_columns() == 0) { return std::vector<packed_table>(); }
+    if (is_empty) { return make_empty_packed_table(); }
+    std::vector<packed_table> result;
+    result.reserve(num_partitions);
+    std::vector<column_view> cols;
+    cols.reserve(input.num_columns());
+
+    auto& h_dst_buf_info = partition_buf_size_and_dst_buf_info->h_dst_buf_info;
+    auto& h_dst_bufs     = src_and_dst_pointers->h_dst_bufs;
+
+    auto cur_dst_buf_info = h_dst_buf_info;
+    detail::metadata_builder mb(input.num_columns());
+
+    for (std::size_t idx = 0; idx < num_partitions; idx++) {
+      // traverse the buffers and build the columns.
+      cur_dst_buf_info = build_output_columns(input.begin(),
+                                              input.end(),
+                                              cur_dst_buf_info,
+                                              std::back_inserter(cols),
+                                              h_dst_bufs[idx],
+                                              mb);
+
+      // pack the columns
+      result.emplace_back(packed_table{
+        cudf::table_view{cols},
+        packed_columns{std::make_unique<std::vector<uint8_t>>(mb.build()),
+                       std::make_unique<rmm::device_buffer>(std::move(out_buffers[idx]))}});
+
+      cols.clear();
+      mb.clear();
+    }
+
+    return result;
+  }
+
+  cudf::table_view const input;
```

Comment:
These should be documented using the standard doxygen style for members, see [the example in our dev docs](https://docs.rapids.ai/api/libcudf/stable/md_developer_guide_documentation). For the members with very long descriptions maybe use a doxygen-style short description and a separate comment for the longer explanation.

Reply:
Added to `chunked_iteration_state` and to `contiguous_split_state`. I didn't add for the other structs in the anonymous namespace for now, but can go back to it if all member should get this treatment.